.github/workflows/check-external-contributor.yml

```
@@ -0,0 +1,14 @@
+name: External Fork check
+on:
+  pull_request: {}
+  
+jobs:
+  check-external:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check PR and label
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: if ${{ github.event.pull_request.head.repo.fork}}=='false'; then gh pr edit $GITHUB_HEAD_REF --add-label "external"; fi 
```

```suggestion
        run: |
        if [ '${{ github.event.pull_request.head.repo.fork}}'=='false' ]; then 
            gh pr edit $GITHUB_HEAD_REF --add-label "external"; 
        fi 
```
Functionally, this is the same, but let's use multi-line yaml for bash statements where possible mainly to keep things readable :+1: